sympy/printing/llvmjitcode.py

```diff
@@ -0,0 +1,192 @@
+        self.int_type = ll.IntType(64)
+
+    def _print_Number(self, n, **kwargs):
+        return ll.Constant(self.fp_type, float(n))
+
+    def _print_Integer(self, expr):
+        return ll.Constant(self.fp_type, float(expr.p))
+
+    def _print_Symbol(self, s):
+        # look up parameter with name s
+        return self.param_dict.get(str(s))
+
+    def _print_Pow(self, expr):
+        base0 = self._print(expr.base)
+        if expr.exp == S.NegativeOne:
+            return self.builder.fdiv(ll.Constant(self.fp_type, 1.0), base0)
+        if expr.exp == S.Half:
+            fn_type = ll.FunctionType(self.fp_type, [self.fp_type])
+            fn = ll.Function(self.module, fn_type, "sqrt")
+            return self.builder.call(fn, [base0], "sqrt")
+        if expr.exp == 2:
+            return self.builder.fmul(base0, base0)
+
+        exp0 = self._print(expr.exp)
+        fn_type = ll.FunctionType(self.fp_type, [self.fp_type, self.fp_type])
+        fn = ll.Function(self.module, fn_type, "pow")
+        return self.builder.call(fn, [base0, exp0], "pow")
+
+    def _print_Mul(self, expr):
+        nodes = [self._print(a) for a in expr.args]
+        e = nodes[0]
+        for node in nodes[1:]:
+            e = self.builder.fmul(e, node)
+        return e
+
+    def _print_Add(self, expr):
+        nodes = [self._print(a) for a in expr.args]
+        e = nodes[0]
+        for node in nodes[1:]:
+            e = self.builder.fadd(e, node)
+        return e
+
+    # TODO - assumes all called functions take one double precision argument.
+    #        Should have a list of math library functions to validate this.
+    def _print_Function(self, expr):
+        name = expr.func.__name__
+        e0 = self._print(expr.args[0])
+        fn_type = ll.FunctionType(self.fp_type, [self.fp_type])
+        fn = ll.Function(self.module, fn_type, name)
+        return self.builder.call(fn, [e0], name)
+
+# ensure lifetime of the execution engine persists (else call to compiled
+#   function will seg fault)
+exe_engines = []
+
+# ensure names for generated functions are unique
+link_names = set()
+current_link_suffix = 0
+
+
+def llvm_jit_code(expr, func_args=None):
+    """Create a native code function from a Sympy expression"""
+
+    global exe_engines, current_link_suffix
+    module = ll.Module('mod1')
+    lj = LLVMJitPrinter(func_args=func_args)
+
+    fp_type = ll.DoubleType()
+    arg_types = []
+    for arg in func_args:
+        arg_type = fp_type
+        arg_types.append(arg_type)
+
+    default_link_name = 'jit_func'
+    current_link_suffix += 1
+    link_name = default_link_name + str(current_link_suffix)
+    link_names.add(link_name)
+
+    fn_type = ll.FunctionType(fp_type, arg_types)
+    fn = ll.Function(module, fn_type, name=link_name)
+    lj.module = module
+    lj.param_dict = {}
+    for i, a in enumerate(func_args):
+        name = str(a)
+        fn.args[i].name = name
+        lj.param_dict[name] = fn.args[i]
```

Comment:
This particular dict is for mapping the Sympy symbol in the expression to the LLVM function argument (accessed by indexed position - `fn.args[i]`) for proper replacement while building the LLVM tree.  I can change it.